src/main/java/seedu/address/logic/parser/UpdateCommandParser.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CANCEL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NEW;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PROCEED;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.update.UpdateCancelCommand;
+import seedu.address.logic.commands.update.UpdateCommand;
+import seedu.address.logic.commands.update.UpdateNewCommand;
+import seedu.address.logic.commands.update.UpdateProceedCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.property.status.Offer;
+
+public class UpdateCommandParser implements Parser<UpdateCommand> {
+
+    private static final String UPDATE_STRING_REGEX = "\\s([1-9]\\d*){1}\\s((new/)|(proceed/)|(cancel/)){1}(?<amount>"
+            + Offer.VALIDATION_REGEX + ")?";
+    private static final Pattern UPDATE_STRING_FORMAT = Pattern.compile(UPDATE_STRING_REGEX,
+            Pattern.CASE_INSENSITIVE);
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the UpdateCommand
+     * and returns an UpdateCommand object for execution.
+     * @throws ParseException If the user input does not conform the expected format.
+     */
+    public UpdateCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NEW, PREFIX_PROCEED, PREFIX_CANCEL);
```

Comment:
Didn't realize that the update command is called as such `update 1 proceed/`, without anything behind the prefixes. So they are not really prefixes in this case, so maybe the forward slashes are not intuitive for the user and should be removed. Sorry!!

An alternative suggestion may be to change the command format to `update INDEX s/STATUS`? 
E.g. `update 1 s/NEW`, `update 1 s/proceed`, `update 1 s/cancel`
In this case, you can even make use of the `ArgumentTokenizer` and `ArgumentMultiMap` classes to parse the inputs.
Will this be better?

Reply:
oh yeah! thats a good idea!

Reply:
Revamped the prefixes!